### PR TITLE
Check license level before install

### LINKED ISSUE #1015: vic-machine: license check
To have serial port working, we need Enterprise license for ESXi host, and Enterprise Plus license for VC. 
vic-machine create should check host and VC license, if not the license required, warning user.

### PR DESCRIPTION
**Pull request by andrewtchin**
Fixes #1015 

```
INFO[2016-06-27T12:56:24-07:00] ### Installing VCH ####                      
INFO[2016-06-27T12:56:24-07:00] Generating certificate/key pair - private key in ./chin-vic-46-key.pem 
INFO[2016-06-27T12:56:26-07:00] Validating supplied configuration            
INFO[2016-06-27T12:56:27-07:00] Firewall status: DISABLED on /Datacenter/host/Austin/patty.eng.vmware.com 
INFO[2016-06-27T12:56:27-07:00] Firewall configuration OK on hosts:          
INFO[2016-06-27T12:56:27-07:00]   /Datacenter/host/Austin/patty.eng.vmware.com 
INFO[2016-06-27T12:56:27-07:00] License check OK on hosts:                   
INFO[2016-06-27T12:56:27-07:00]   /Datacenter/host/Austin/patty.eng.vmware.com 
```

```
INFO[2016-06-27T12:53:08-07:00] ### Installing VCH ####                      
INFO[2016-06-27T12:53:08-07:00] Generating certificate/key pair - private key in ./chin-vic-32-key.pem 
INFO[2016-06-27T12:53:09-07:00] Validating supplied configuration            
INFO[2016-06-27T12:53:09-07:00] Firewall status: DISABLED on /ha-datacenter/host/localhost.localdomain/localhost.localdomain 
INFO[2016-06-27T12:53:09-07:00] Firewall configuration OK on hosts:          
INFO[2016-06-27T12:53:09-07:00]   /ha-datacenter/host/localhost.localdomain/localhost.localdomain 
ERRO[2016-06-27T12:53:09-07:00] License check FAILED:                        
ERRO[2016-06-27T12:53:09-07:00]   Host license missing feature 'serialuri'   
ERRO[2016-06-27T12:53:09-07:00] License does not meet minimum requirements to use VIC 
ERRO[2016-06-27T12:53:09-07:00] Creation cannot continue: configuration validation failed 
```

### CODE REVIEW

**Comment by dougm** [comment]
lgtm

File: lib/install/validate/validator.go [hunk 1]
```diff
@@ -490,6 +492,130 @@ func (v *Validator) firewall(ctx context.Context) {
...
+	if hosts, err = v.Session.Datastore.AttachedClusterHosts(ctx, v.Session.Cluster); err != nil {
+		log.Errorf("Unable to get the list of hosts attached to given storage: %s", err)
+		return err
+	}
+
+	lm := license.NewManager(client)
+
+	am, err := lm.AssignmentManager(ctx)
+	if err != nil {
+		v.NoteIssue(err)
```
Comment (emlin) [inline lib/install/validate/validator.go]: Looks like you noteIssue here and outside of this method.

Comment (andrewtchin) [inline lib/install/validate/validator.go]: good catch, fixed

**Comment by emlin** [comment]
LGTM

**Comment by andrewtchin** [comment]
♻️

**Pull request merged**